cmd/cleanup-controller/handlers/admission/resource/handlers.go

```diff
@@ -6,13 +6,23 @@ import (
 	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/kyverno/kyverno/pkg/auth/checker"
+	manager "github.com/kyverno/kyverno/pkg/controllers/ttl"
 	admissionutils "github.com/kyverno/kyverno/pkg/utils/admission"
 	validation "github.com/kyverno/kyverno/pkg/validation/resource"
 	"github.com/kyverno/kyverno/pkg/webhooks/handlers"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 func Validate(ctx context.Context, logger logr.Logger, request handlers.AdmissionRequest, _ time.Time) handlers.AdmissionResponse {
+	checker := manager.GetChecker()
 	metadata, _, err := admissionutils.GetPartialObjectMetadatas(request.AdmissionRequest)
+	gvr := admissionutils.GetGVR(request.AdmissionRequest)
+
+	if !hasResourcePermissions(logger, gvr, checker) {
+		logger.Info("resource doesn't have required permissions for deletion ", gvr)
```

Comment:
Nope, it's not the correct way to use a logger...

Reply:
do I have to divide the logging parts in key value pairs right?

Reply:
Yes, this is not correct.